src/colorizer/FrameCache.ts

```diff
@@ -1,48 +1,52 @@
-import { Texture } from "three";
-
-type CacheEntry = {
-  frame: Texture;
-  index: number;
-  prev: CacheEntryNullable;
-  next: CacheEntryNullable;
+type DisposableValue = {
+  dispose?: () => void;
 };
 
-type CacheEntryNullable = CacheEntry | null;
+type CacheEntry<E extends DisposableValue> = {
+  key: string;
+  value: E;
+  size: number;
+  prev: CacheEntry<E> | null;
+  next: CacheEntry<E> | null;
+};
 
-/** LRU cache for textures. Calls `dispose` on eviction to keep GPU memory under control. */
-export default class FrameCache {
-  private data: CacheEntryNullable[];
-  private first: CacheEntryNullable;
-  private last: CacheEntryNullable;
-  private numItems: number;
+/**
+ * Generic LRU cache for data, intended for Texture or other GPU resources.
+ * Calls `dispose` on eviction to keep GPU memory under control.
+ */
+export default class FrameCache<E extends DisposableValue> {
+  private data: Map<string, CacheEntry<E>>;
```

Comment:
Now uses string keys instead of integers which index into an array, which allows more flexible data to be stored (e.g. features)